packages/react-meteor-data/suspense/useTracker.ts

```diff
@@ -0,0 +1,305 @@
+  error?: unknown
+}
+
+// Used to create a forceUpdate from useReducer. Forces update by
+// incrementing a number whenever the dispatch method is invoked.
+const fur = (x: number): number => x + 1
+const useForceUpdate = () => useReducer(fur, 0)[1]
+
+export type IReactiveFn<T> = (c?: Tracker.Computation) => Promise<T>
+
+export type ISkipUpdate<T> = <T>(prev: T, next: T) => boolean
+
+interface TrackerRefs {
+  computation?: Tracker.Computation
+  isMounted: boolean
+  trackerData: any
+}
+
+function resolveAsync(key: string, promise: Promise<unknown>, deps: DependencyList = []) {
+  const cached = cache.get(key) ?? false
+  useEffect(() => {
+    return () => {
+      setTimeout(() => {
+        // maybe verify deps?
+        if (cached) cache.delete(key)
+      }, 0)
+    }
+  }, [cached, key, ...deps])
+  if (promise === null) return null
+  if (cached) {
+    if ('error' in cached) throw cached.error
+    if ('result' in cached) return cached.result
+    throw cached.promise
+  }
+
+  const entry: Cache = {
+    deps,
+    promise: new Promise((resolve, reject) => {
+      promise
+        .then((result: any) => {
+          entry.result = result
+          resolve(result)
+        })
+        .catch((error: any) => {
+          entry.error = error
+          reject(error)
+        })
+        .finally(() => {
+          // if 0 recursion :/
+          // also forcing an update thigs when out of hand
+          setTimeout(() => {
+            cache.delete(key)
+          }, 10)
+        })
+    })
+  }
+  cache.set(key, entry)
+  throw entry.promise
+}
+
+export function useTrackerNoDeps<T = any>(key: string, reactiveFn: IReactiveFn<T>, skipUpdate: ISkipUpdate<T> = null): T {
+  const { current: refs } = useRef<TrackerRefs>({
+    isMounted: false,
+    trackerData: null
+  })
+  const forceUpdate = useForceUpdate()
+
+  // Without deps, always dispose and recreate the computation with every render.
+  if (refs.computation != null) {
+    refs.computation.stop()
+    // @ts-expect-error This makes TS think ref.computation is "never" set
+    delete refs.computation
+  }
+
+  // Use Tracker.nonreactive in case we are inside a Tracker Computation.
+  // This can happen if someone calls `ReactDOM.render` inside a Computation.
+  // In that case, we want to opt out of the normal behavior of nested
+  // Computations, where if the outer one is invalidated or stopped,
+  // it stops the inner one.
+  Tracker.nonreactive(() =>
+    Tracker.autorun((c: Tracker.Computation) => {
+      refs.computation = c
+
+      const data: Promise<any> = Tracker.withComputation(c, async () => await reactiveFn(c))
+      if (c.firstRun) {
+        // Always run the reactiveFn on firstRun
+        refs.trackerData = data
+        /// TODO : this about this case
+      } else if (!skipUpdate || !skipUpdate(refs.trackerData, data)) {
+        // For any reactive change, forceUpdate and let the next render rebuild the computation.
+        forceUpdate()
```

Comment:
I think this should be called only after the `promise` resolves, but I'm not sure. This'd need to be tested at least in three ways:
1. `async` computation without `await`.
2. `async` computation with `await` and a reactive source _before_ `await`.
3. `async` computation with `await` and a reactive source _after_ `await`.

Reply:
so for the tests I got the following results:

1. using the following code:

```js
  const docs = useTrackerSuspense('name', () => active ? LinksCollection.find().fetchAsync() : null, [active]);
  const docs2 = useTracker(() => active ? LinksCollection.find().fetch() : null, [active]);

  console.log({
    docs,
    docs2
  });
```
it was always reactive, every time there is a new link it would add to the docs variable.

2. using the following code:

```js
  const docs = useTrackerSuspense('name', async () => {
    const d = await LinksCollection.find().fetchAsync()
    const placeholders = await fetch('https://jsonplaceholder.typicode.com/todos').then(x => x.json())
    return active ? d : null
  }, [active]);
  const docs2 = useTracker(() => active ? LinksCollection.find().fetch() : null, [active]);

  console.log({
    docs,
    docs2
  });
```
it was always reactive, every time there is a new link it would add to the docs variable.

3. using the following code:

```js
 const docs = useTrackerSuspense('name', async () => {
    const placeholders = await fetch('https://jsonplaceholder.typicode.com/todos').then(x => x.json())
    const d = await LinksCollection.find().fetchAsync()
    return active ? d : null
  }, [active]);
  const docs2 = useTracker(() => active ? LinksCollection.find().fetch() : null, [active]);

  console.log({
    docs,
    docs2
  });

```

It was not reactive and did not change the docs variable, even when you wrap it with a Tracker.withComputation, it still needs to become reactive, and in our case, it does not .

Reply:
1. and 2. are OK then. As for the 3., I guess we'd need to make sure it works with `Tracker.withComputation` - I think it should. Let's discuss it tomorrow.